Libraries/Opc.Ua.Client/Session.cs

```diff
@@ -5264,6 +5265,26 @@ public bool ResendData(IEnumerable<Subscription> subscriptions, out IList<Servic
                     !String.IsNullOrEmpty(identityPolicy.SecurityPolicyUri);
             }
         }
+        /// <summary>
+        /// Validates the ServerCertificate ApplicationUri to match the ApplicationUri of the Endpoint for an open call (Spec Part 4 5.4.1)
+        /// </summary>
+        private void ValidateServerCertificateApplicationUri(
+            X509Certificate2 serverCertificate)
+        {
+            var applicationUri = m_endpoint.Description.Server.ApplicationUri;
+            //check is only neccessary if the ApplicatioUri is specified for the Endpoint
```

Comment:
I think there must be a ApplicationUri defined, or an exception should be thrown.
Also check like this: m_endpoint?.Description?.Server?.ApplicationUri to catch all null cases.